lib/CMakeLists.txt

```diff
@@ -19,7 +19,13 @@ set( CONFIG_PACKAGE_INSTALL_DIR ${LIB_INSTALL_DIR}/cmake/hcc )
 add_mcwamp_library(mcwamp mcwamp.cpp)
 add_mcwamp_library(mcwamp_atomic mcwamp_atomic.cpp)
 
-target_link_libraries(mcwamp dl)
+# Library interface to use runtime
+add_library(hccrt INTERFACE)
+target_link_libraries(hccrt INTERFACE "-Wl,-rpath $<TARGET_FILE_DIR:$<TARGET_NAME:mcwamp>>" "-Wl,--whole-archive" mcwamp "-Wl,--no-whole-archive")
+target_link_libraries(hccrt INTERFACE dl)
+target_link_libraries(hccrt INTERFACE c++)
+target_link_libraries(hccrt INTERFACE c++abi)
```

Comment:
@scchan could you comment on how these lines be changed if we enable libstdc++ as the default C++ runtime for HCC runtime? would it be just an `if(USE_LIBCXX)` check?